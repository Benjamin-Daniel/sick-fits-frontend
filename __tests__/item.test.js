import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const fakeItem = {
  id: 'ahc343',
  title: 'A Cool Item',
  price: '4000',
  description: 'A very cool item',
  image: 'dog.jpg',
  largeImage: 'large-dog.jpg',
};
describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  // it('renders image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title)

  // });


  // it('renders and pricetag and title properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.children().text()).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });

  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link').exists()).toBe(true);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  //   console.log(buttonList.debug());
  // });


});
