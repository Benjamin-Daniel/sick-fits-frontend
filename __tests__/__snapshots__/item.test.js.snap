// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ahc343",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    A very cool item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ahc343",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ahc343"
    />
    <DeleteItem
      id="ahc343"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
